### PR TITLE
Added example for InMemory exporter

### PR DESCRIPTION
**Pull request by utpilla**
## Changes

Please provide a brief description of the changes here.
- Added example for InMemoryExporter in Eaxmples.Console project
- Updated OpenTelemetry.InMemoryExporter README.md to point to the InMemoryExporter example

### CODE REVIEW

File: examples/Console/Examples.Console.csproj [hunk 1]
```diff
@@ -36,5 +36,6 @@
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry\OpenTelemetry.csproj" />
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Zipkin\OpenTelemetry.Exporter.Zipkin.csproj" />
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Exporter.Jaeger\OpenTelemetry.Exporter.Jaeger.csproj" />
+    <ProjectReference Include="..\..\src\OpenTelemetry.Exporter.InMemory\OpenTelemetry.Exporter.InMemory.csproj" />
```
Comment (reyang) [inline examples/Console/Examples.Console.csproj:39]:
```suggestion
    <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Exporter.InMemory\OpenTelemetry.Exporter.InMemory.csproj" />
```

File: examples/Console/TestInMemoryExporter.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                .AddProcessor(new MyProcessor()) // This must be added before InMemoryExporter
+                .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            // The above line is required only in applications
+            // which decide to use OpenTelemetry.
+
+            // Libraries would simply write the following lines of code to
+            // emit activities, which are the .NET representation of OpenTelemetry Spans.
+            var source = new ActivitySource("MyCompany.MyProduct.MyWebServer");
```
Comment (reyang) [inline examples/Console/TestInMemoryExporter.cs:52]: Most of these logic seems to be duplicated with other exporter examples? Is there a way to de-dupe?

Comment (cijothomas) [reply examples/Console/TestInMemoryExporter.cs:52]: https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/examples/Console/InstrumentationWithActivitySource.cs we could move to this. Its already used by otlp and other exporters, so perhaps a good idea to move everything to use this.
(the docs/trace/getting-started can keep the absolute simple example)

Comment (utpilla) [reply examples/Console/TestInMemoryExporter.cs:52]: Should we add InMemoryExporter to the docs/trace/getting-started example?

Comment (cijothomas) [reply examples/Console/TestInMemoryExporter.cs:52]: No.

File: examples/Console/TestConsoleExporter.cs [hunk 3]
```diff
@@ -31,94 +31,32 @@ internal class TestConsoleExporter
...
+            return RunWithActivitySource();
+        }
+
+        private static object RunWithActivitySource()
+        {
+            // Enable OpenTelemetry for the sources "Samples.SampleServer" and "Samples.SampleClient"
+            // and use Console exporter.
+            using var openTelemetry = Sdk.CreateTracerProviderBuilder()
+                    .AddSource("Samples.SampleClient", "Samples.SampleServer")
+                    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("otlp-test"))
```
Comment (cijothomas) [inline examples/Console/TestConsoleExporter.cs:43]:
```suggestion
                    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("console-test"))
```

File: examples/Console/TestInMemoryExporter.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    .Build();
+
+            // The above line is required only in applications
+            // which decide to use OpenTelemetry.
+            using (var sample = new InstrumentationWithActivitySource())
+            {
+                sample.Start();
+
+                System.Console.WriteLine("Traces are being created and exported " +
+                    "to Console in the background. " +
```
Comment (cijothomas) [inline examples/Console/TestInMemoryExporter.cs:66]: fix the text.

File: examples/Console/TestConsoleExporter.cs (hunk 3 above)
Comment (utpilla) [reply examples/Console/TestConsoleExporter.cs]: Fixed this.

File: examples/Console/TestInMemoryExporter.cs (hunk 4 above)
Comment (utpilla) [reply examples/Console/TestInMemoryExporter.cs]: Fixed this.

**Review by cijothomas — APPROVED** [review]

**Pull request closed without merging**